test/cluster/dtest/audit_test.py

```diff
@@ -232,20 +255,29 @@ def update_audit_settings(self, audit_settings, modifiers=None):
     @override
     def get_audit_log_list(self, session, consistency_level):
         lines = self.unix_socket_listener.get_lines()
+        entries = []
+        for idx, line in enumerate(lines):
+            entries.append(self.line_to_row(line, idx))
+        return entries
 
-        return lines
-
-    def line_to_row(self, line):
+    def line_to_row(self, line, idx):
         metadata, data = line.split(": ", 1)
-        elems = list(map(lambda x: x[1:], data.split('", ')))
-        elems[-1] = elems[-1][:-1]
-        date = datetime.datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+        data = "".join(data.splitlines()) # Remove newlines
+        fields = ["node", "category", "cl", "error", "keyspace", "query", "client_ip", "table", "username"]
+        regexp = r""
+        for field in fields:
+            comma = ""
+            if field != fields[-1]:
+                comma = ", " # require comma if not the last field
+            regexp += f"{field}=\"(?P<{field}>.*)\"{comma}"
+        match = re.match(regexp, data)
 
-        node = elems[0].split(":")[0]
-        source = elems[6].split(":")[0]
-        # static uuid is used to keep the same format as the table audit
-        event_time = uuid.UUID("2e4bc246-fea1-11ee-b73f-51b0926539af")
-        t = self.named_tuple_factory(date, node, event_time, elems[1], elems[2], elems[3] == "true", elems[4], elems[5], source, elems[7], elems[8])
+        date = datetime.datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+        node = match.group("node").split(":")[0]
+        statement = match.group("query").replace("\\", "") 
+        source = match.group("client_ip").split(":")[0]
+        event_time = uuid.UUID(int=idx)
+        t = self.named_tuple_factory(date, node, event_time, match.group("category"), match.group("cl"), match.group("error") == "true", match.group("keyspace"), statement, source, match.group("table"), match.group("username"))
```

Comment:
knowing python magic you probably can also reuse fields list and generate this dynamically as regex above